### PR TITLE
Fix extensions loading and parsing

### PR DESCRIPTION
**Pull request by lephilousophe**
This fixes issue #446 by being more flexible when parsing extensions.
If the array version is available, it is used, else, it falls back on space separated string.

In addition, if the implementation doesn't have any extension, parsing doesn't fail anymore and if malloc fails, extensions parsing fails.

### CODE REVIEW

**Review by Dav1dde — COMMENTED** [review]
Looks good, actually gets rid of some complexity!

I left one comment which, I think, might lead us to `free` random addresses.

File: glad/generator/c/templates/gl.c [hunk 1]
```diff
@@ -36,36 +36,24 @@ static void _post_call_{{ feature_set.name }}_callback_default(void *ret, const
...
         unsigned int num_exts_i = 0;
         char **exts_i = NULL;
-        if ({{ 'glGetStringi'|ctx }} == NULL || {{ 'glGetIntegerv'|ctx }} == NULL) {
-            return 0;
-        }
         {{ 'glGetIntegerv'|ctx }}(GL_NUM_EXTENSIONS, (int*) &num_exts_i);
-        if (num_exts_i > 0) {
-            exts_i = (char **) malloc(num_exts_i * (sizeof *exts_i));
-        }
+        exts_i = (char **) malloc((num_exts_i + 1) * (sizeof *exts_i));
```
[the remark this answers is not in the document]
Comment (lephilousophe) [reply glad/generator/c/templates/gl.c:56]: Indeed, `calloc` wouldn't fit here if we strictly follow the standards as `NULL` is equal to `0` but that doesn't means `0` is all-zero bits.

But I don't see where `free` can be used on uninitialized content.
When `glad_gl_free_extensions` is called from `glad_gl_find_extensions_XXX`, the array is fully initialized and `NULL` terminated.
In `glad_gl_get_extensions`, the array is progressively initialized and, if the loop would end up early, the array is properly `NULL` terminated before calling `glad_gl_free_extensions`.
That will make `glad_gl_free_extensions` free all initialized values and stop before hitting uninitialized ones.

[the remark this answers is not in the document]
Comment (Dav1dde) [reply glad/generator/c/templates/gl.c:56]: > In glad_gl_get_extensions, the array is progressively initialized and, if the loop would end up early, the array is properly NULL terminated before calling glad_gl_free_extensions.

I missed that, thanks.

**Comment by lephilousophe** [comment]
I fixed the test failure (superfluous comma) and all tests now build fine.

**Pull request merged**